samples/containers/sample-list.js

```diff
@@ -1,17 +1,36 @@
 const React = require('react')
 const { connect } = require('react-redux')
 const { Link } = require('react-router')
+const FileDrop = require('react-file-drop')
+const { forEach } = require('lodash')
 
 const { getSampleListProps } = require('../getters')
-const { loadSampleList } = require('../actions')
+const { loadSampleList, createSample } = require('../actions')
 
 class SampleListContainer extends React.Component {
+  onFilesDrop (e) {
+    const { createSample } = this.props
+    const files = e && e.dataTransfer && e.dataTransfer.files
+
+    console.log('file drop', e)
+    if (files) {
+      e.preventDefault()
+      e.stopPropagation()
+      forEach(files, createSample)
+    }
+  }
+
   render () {
-    const { sampleList, isLoading } = this.props
+    const { sampleList, isLoading, isCreating, error } = this.props
 
     return <div>
+      <FileDrop
+        frame={document}
+        onFrameDrop={this.onFilesDrop.bind(this)} />
       <header>
-        samples are {isLoading ? 'loading' : 'here'}
+        <div>samples are {isLoading ? 'loading' : 'here'}</div>
+        {isCreating && <div>'creating samples…'</div>}
```

Comment:
store whether or not the sample is being created or analyzed somewhere in the store, then in the getters attach this information to the samples and then `analyzingSamples: [sample1, sample2, ...]` is just a filtered list where `analyzing: true`.